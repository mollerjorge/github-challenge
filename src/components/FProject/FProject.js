import React from "react";
import PropTypes from "prop-types";
import FProjectHolder from "./FProject.style.js";
import ContributorsList from "../Contributor/ContributorsList";
import { Icon } from "antd";

const FProject = ({
	project: { name, open_issues_count, forks_count, description },
	contributors
}) => {
	return (
		<FProjectHolder>
			<div
				style={{
					display: "flex",
					justifyContent: "space-between",
					alignItems: "center"
				}}
			>
				<h1>{name}</h1>
				<div>
					<span>
						<Icon type="warning" />&nbsp;{open_issues_count}
					</span>
					&nbsp;&nbsp;
					<span>
						<Icon type="fork" />&nbsp;{forks_count}
					</span>
				</div>
			</div>
			<p>{description}</p>
			<hr />
			<ContributorsList contributors={contributors} />
		</FProjectHolder>
	);
};

FProject.propTypes = {
	project: PropTypes.shape({
		name: PropTypes.string,
		open_issues_count: PropTypes.number,
		forks_count: PropTypes.number,
		description: PropTypes.string
	}),
	contributors: PropTypes.array.isRequired
};

export default FProject;
