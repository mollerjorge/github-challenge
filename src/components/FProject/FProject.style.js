import styled from "styled-components";

const FProejctHolder = styled.div`
	min-height: 100vh
	background: #f9f9f9;
	padding: 2rem;

	hr {
		opacity: .3;
	}

	.contributors-list {
		padding: 0;
	}
`;

FProejctHolder.displayName = "FProejctHolder";

export default FProejctHolder;
