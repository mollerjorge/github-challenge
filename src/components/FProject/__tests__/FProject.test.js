import { shallow } from "enzyme";
import React from "react";
import FProject from "../FProject";

describe("FProject", () => {
	const contributorsMock = [
		{
			login: "george",
			avatar_url: "http://mockurl.com",
			contributions: 4
		},
		{
			login: "camille",
			avatar_url: "http://mockurl.com",
			contributions: 5
		}
	];
	const projectMock = {
			
				name: "this is a mock project",
				open_issues_count: 4,
				forks_count: 10,
				description: "this is a mock description"
			
		};
	const fproject = shallow(<FProject project={projectMock} contributors={contributorsMock} />);

	it("renders correctly", () => {
		expect(fproject).toMatchSnapshot();
	});

	it('displays ui elements correctly', () => {
		expect(fproject.find('FProejctHolder').find('h1').text()).toEqual('this is a mock project');
		expect(fproject.find('FProejctHolder').find('p').text()).toEqual('this is a mock description');
	});

	it('displays the ContributorsList Component', () => {
		expect(fproject.find('ContributorsList').length).toEqual(1);
	});
});
