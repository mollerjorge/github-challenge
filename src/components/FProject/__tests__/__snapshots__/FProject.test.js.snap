// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FProject renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FProject
    contributors={
        Array [
            Object {
              "avatar_url": "http://mockurl.com",
              "contributions": 4,
              "login": "george",
            },
            Object {
              "avatar_url": "http://mockurl.com",
              "contributions": 5,
              "login": "camille",
            },
          ]
    }
    project={
        Object {
            "description": "this is a mock description",
            "forks_count": 10,
            "name": "this is a mock project",
            "open_issues_count": 4,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          style={
                    Object {
                              "alignItems": "center",
                              "display": "flex",
                              "justifyContent": "space-between",
                            }
          }
>
          <h1>
                    this is a mock project
          </h1>
          <div>
                    <span>
                              <Icon
                                        type="warning"
                              />
                               
                              4
                    </span>
                      
                    <span>
                              <Icon
                                        type="fork"
                              />
                               
                              10
                    </span>
          </div>
</div>,
        <p>
          this is a mock description
</p>,
        <hr />,
        <ContributorsList
          contributors={
                    Array [
                              Object {
                                "avatar_url": "http://mockurl.com",
                                "contributions": 4,
                                "login": "george",
                              },
                              Object {
                                "avatar_url": "http://mockurl.com",
                                "contributions": 5,
                                "login": "camille",
                              },
                            ]
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              this is a mock project
</h1>,
            <div>
              <span>
                            <Icon
                                          type="warning"
                            />
                             
                            4
              </span>
                
              <span>
                            <Icon
                                          type="fork"
                            />
                             
                            10
              </span>
</div>,
          ],
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "this is a mock project",
            },
            "ref": null,
            "rendered": "this is a mock project",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  <Icon
                                    type="warning"
                  />
                   
                  4
</span>,
                "  ",
                <span>
                  <Icon
                                    type="fork"
                  />
                   
                  10
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Icon
                      type="warning"
/>,
                    " ",
                    4,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "type": "warning",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " ",
                  4,
                ],
                "type": "span",
              },
              "  ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Icon
                      type="fork"
/>,
                    " ",
                    10,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "type": "fork",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " ",
                  10,
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "this is a mock description",
        },
        "ref": null,
        "rendered": "this is a mock description",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "contributors": Array [
            Object {
              "avatar_url": "http://mockurl.com",
              "contributions": 4,
              "login": "george",
            },
            Object {
              "avatar_url": "http://mockurl.com",
              "contributions": 5,
              "login": "camille",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            style={
                        Object {
                                    "alignItems": "center",
                                    "display": "flex",
                                    "justifyContent": "space-between",
                                  }
            }
>
            <h1>
                        this is a mock project
            </h1>
            <div>
                        <span>
                                    <Icon
                                                type="warning"
                                    />
                                     
                                    4
                        </span>
                          
                        <span>
                                    <Icon
                                                type="fork"
                                    />
                                     
                                    10
                        </span>
            </div>
</div>,
          <p>
            this is a mock description
</p>,
          <hr />,
          <ContributorsList
            contributors={
                        Array [
                                    Object {
                                      "avatar_url": "http://mockurl.com",
                                      "contributions": 4,
                                      "login": "george",
                                    },
                                    Object {
                                      "avatar_url": "http://mockurl.com",
                                      "contributions": 5,
                                      "login": "camille",
                                    },
                                  ]
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                this is a mock project
</h1>,
              <div>
                <span>
                                <Icon
                                                type="warning"
                                />
                                 
                                4
                </span>
                  
                <span>
                                <Icon
                                                type="fork"
                                />
                                 
                                10
                </span>
</div>,
            ],
            "style": Object {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "space-between",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "this is a mock project",
              },
              "ref": null,
              "rendered": "this is a mock project",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    <Icon
                                        type="warning"
                    />
                     
                    4
</span>,
                  "  ",
                  <span>
                    <Icon
                                        type="fork"
                    />
                     
                    10
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Icon
                        type="warning"
/>,
                      " ",
                      4,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "type": "warning",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " ",
                    4,
                  ],
                  "type": "span",
                },
                "  ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <Icon
                        type="fork"
/>,
                      " ",
                      10,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "type": "fork",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " ",
                    10,
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "this is a mock description",
          },
          "ref": null,
          "rendered": "this is a mock description",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "contributors": Array [
              Object {
                "avatar_url": "http://mockurl.com",
                "contributions": 4,
                "login": "george",
              },
              Object {
                "avatar_url": "http://mockurl.com",
                "contributions": 5,
                "login": "camille",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
